killerbee/pcapdump.py

```diff
@@ -103,27 +106,24 @@ def pcap_next(self):
 
 
 class PcapDumper:
-    def __init__(self, datalink, savefile, ppi = False):
+    def __init__(self, datalink, savefile, ppi = False, folder='.'):
         '''
         Creates a libpcap file using the specified datalink type.
         @type datalink: Integer
         @param datalink: Datalink type, one of DLT_* defined in pcap-bpf.h
-        @type savefile: String or file-like object
-        @param savefile: Output libpcap filename to open, or file-like object
-        @type ppi: Boolean
-        @param ppi: Include CACE Per-Packet Information (defaults to False)
+        @type savefile: String
+        @param savefile: Output libpcap filename to open
         @rtype: None
         '''
         if ppi: from killerbee.pcapdlt import DLT_PPI
-        self.ppi = ppi
-
-        if isinstance(savefile, basestring):
-            self.__fh = open(savefile, mode='wb')
-        elif hasattr(savefile, 'write'):
-            self.__fh = savefile
-        else:
-            raise ValueError("Unsupported type for 'savefile' argument")
-
+        self.ppi = ppi        
+        if folder[-1] == "/":
+            folder = folder[:-1]
+        if not os.path.exists(folder):
+            log_message = "PcapDump: Creating directory to store results"
+            logging.debug(log_message)
+            os.makedirs(folder)
+        self.__fh = open(folder+savefile, mode='wb')
```

Comment:
As pcapdump I think is used across a few applications, I can't have this write files all the time. If we want it to do a write of this way, it needs to default to not happening and be a parameter that turns it on.